pkg/segment/structs/evaluationstructs.go

```diff
@@ -240,6 +240,26 @@ type TcOptions struct {
 	OtherStr   string
 }
 
+type BinCmdOptions struct {
+	BinSpanOptions *BinSpanOptions
+	MinSpan        *BinSpanLength
+	MaxBins        uint64
+	Start          *float64
+	End            *float64
+	AlignTime      *uint64
```

Comment:
Why are these pointers?

Reply:
We would need a way to store start and end values and we cannot use default values i.e. 0 because the numerical values in the field could be 0 or negative. Pointers are just used as a concise way to represent if the value is set or not.